worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -334,22 +334,28 @@ thisScriptObject = {
             var component = extractSourceCallback !== undefined
                             ? extractSourceCallback(f)
                             : getCurrentWorkerName(document) + " " + getCurrentWorkerVersion(document);
-            var failureObject = {
+            var isWarningFlag = (typeof isWarning === 'function') ? isWarning(f): false;
+            var errorObject = {
                 ID: f.getFailureId(),
-                STACK: f.getFailureStack() || undefined,
                 WORKFLOW_ACTION: workflowAction,
                 COMPONENT: component,
                 WORKFLOW_NAME: document.getRootDocument().getField("CAF_WORKFLOW_NAME").getStringValues().get(0),
                 MESSAGE: f.getFailureMessage(),
                 DATE: new Date().toISOString(),
                 CORRELATION_ID: document.getCustomData("correlationId") || undefined
             };
-            if (extraFailureFields) {
-                for each (var key in Object.keys(extraFailureFields)) {
-                    failureObject[key] = extraFailureFields[key];
+
+            if (!isWarningFlag) {
```

Comment:
@dermot-hardy do we not want to allow extra failure subfields or the failure stack to be part of the warnings?

Reply:
It's debatable but my recollection fro the storytime is that we said we thought that we didn't need them in the warnings, so the code aligns with that decision as it is.

Reply:
Yes.  In the storytime, it is finalized that the stack and extra failure subfields not needed for warnings.